google/cloud/bigtable/instance_admin_client.h

```diff
@@ -18,321 +18,39 @@
 #include "google/cloud/bigtable/admin/bigtable_instance_admin_connection.h"
 #include "google/cloud/bigtable/client_options.h"
 #include "google/cloud/bigtable/version.h"
-#include <google/bigtable/admin/v2/bigtable_instance_admin.grpc.pb.h>
 #include <memory>
 #include <string>
 
 namespace google {
 namespace cloud {
 namespace bigtable {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
-// Forward declare some classes so we can be friends.
-class InstanceAdmin;
-namespace internal {
-template <typename Client, typename Response>
-class AsyncLongrunningOperation;
-class LoggingInstanceAdminClient;
-}  // namespace internal
 
 /**
- * Connects to Cloud Bigtable's instance administration APIs.
+ * Holds configuration options for `bigtable::InstanceAdmin`.
  *
- * This class is used by the Cloud Bigtable wrappers to access Cloud Bigtable.
- * Multiple `bigtable::InstanceAdmin` objects may share a connection via a
- * single `InstanceAdminClient` object. The `InstanceAdminClient` object is
- * configured at construction time, this configuration includes the credentials,
- * access endpoints, default timeouts, and other gRPC configuration options.
- * This is an interface class because it is also used as a dependency injection
- * point in some of the tests.
+ * This class is used as a conduit to pass configuration options to
+ * `bigtable::InstanceAdmin`. Formerly, this class was responsible for
+ * initiating a connection to the Cloud Bigtable Instance Admin service. Now
+ * that connection is initiated by `bigtable::InstanceAdmin`. This class is
+ * maintained only for backwards compatibility.
  *
- * @par Cost
- * Applications should avoid unnecessarily creating new objects of type
- * `InstanceAdminClient`. Creating a new object of this type typically requires
- * connecting to the Cloud Bigtable servers, and performing the authentication
- * workflows with Google Cloud Platform. These operations can take many
- * milliseconds, therefore applications should try to reuse the same
- * `InstanceAdminClient` instances when possible.
+ * @deprecated Please consider using
```

Comment:
In a future PR, maybe we should add a `[[deprecated]]` attribute (or its equivalent for compilers that do not support that).

Reply:
Ack.